constraint/pkg/client/regolib/src.go

```diff
@@ -0,0 +1,38 @@
+package regolib
+
+const (
+	// Finds all violations for a given target
+	denySrc = `
+package hooks.{{.Target}}
+
+deny[response] {
+	data.hooks.{{.Target}}.library.matching_constraints[constraint]
+	data.templates.{{.Target}}[_].deny[response] with input.constraint as constraint
+}
+`
+
+	// Runs audit on a given target
+	auditSrc = `
+package hooks.{{.Target}}
+
+audit[response] {
+	data.hooks.{{.Target}}.library.matching_reviews_and_constraints[[review, constraint]]
+	cnst_key := ["constraint"]
+	keys := [k | review[k]]
+	all_keys := array.concat(cnst_key, keys)
+	full_review := {k: v | v = audit_field_or_constraint(k, review, constraint); k = all_keys[_]}
+	review.constraint = constraint
+	data.templates.{{.Target}}[_].deny[response] with input as review
```

Comment:
How important is it to merge the constraint into the review? Alternatively we could supply the input document as...
```
input.review     # AdmissionReview object
input.constraint # Constraint object
```

This would also give us a bit of room in the future if we wanted to introduce other fields in the input document without worrying about conflicts.

Reply:
Providing them separate sounds good to me. Per Morgante's comment on the design doc, we may want to choose a more generic name than "review" in that case, as other targets may not have the concept of a review request.